### ISSUE TITLE
SC2120: ShellCheck doesn't process ${1:-default} or ${1-default)

### ISSUE THREAD

**Issue by mech422** [koalaman/shellcheck#1695]
#### For bugs
- Rule 2120
- shellcheck version: 0.7.0
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
MSG_INDENT=4
indent() { MSG_INDENT=$(( MSG_INDENT + ${1:-2} )) ; }

indent


```

#### Here's what shellcheck currently says:
```
Line 3	SC2120: indent references arguments, but none are ever passed.
Line 5	SC2119: Use indent "$@" if function's $1 should mean script's $1.
```
#### Here's what I wanted or expected to see:

TODO: 
  Shellcheck should realize value of 2 is being substituted as $1 isn't set.

**Comment by olorin37** [comment]
Problem is even more demonic, because using `${1:=default}` makes shellcheck happy, but it breaks bash code, because bash do not allow for assigning to the arguments (`$1`, `$2` ... `$@`). So, now shellcheck encourages to use not valid bash code.

```
#!/usr/bin/env bash

# Problematic case #

fun_ok() {  # <- shellcheck warns here 2120
  local var=${1:-default_value}
  echo "$var"
}

fun_ok # <- and here 2119

# Even more problematic #

fun_bash_runtime_error() {
  local var=${1:=default_value}  #  <- now shellcheck is quiet
              # ^ cannot assign to $1 or $@
  echo "$var"
}

fun_bash_runtime_error

# Current solution #

# shellcheck disable=SC2120
fun_and_shellcheck_ok() {
  local var=${1:-default_value}
  echo "$var"
}

fun_and_shellcheck_ok
```

So, in my opinion it is important to eliminate second case, and then do not warn in first case. By the way shellcheck should warn in each case when someone try to assign argument with parameter expansion (but this should be probably another issue).

**Comment by leagris** [comment]
Maybe there could be a severe SC code for when an argument is made explicitly mandatory with:
```sh
${1:?}
```
